### PR TITLE
nf-test: ensure makeWrapper is restricted to nativeBuildInputs

### PR DESCRIPTION
**Pull request by SFrijters**
## Description of changes

Fix one part of `error: makeWrapper/makeShellWrapper must be in nativeBuildInputs` - ~~currently the `nextflow` derivation itself still will put makeWrapper on the PATH.~~ Fixed by https://github.com/NixOS/nixpkgs/pull/339197

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SFrijters** [comment]
@rollf Related to https://github.com/NixOS/nixpkgs/pull/339197#discussion_r1758461967 - feel free to use this in your other PR if you think it makes sense. Currently testing the build with `nix build .#pkgsCross.aarch64-multiplatform.nf-test -L` with a similar fix implemented for `nextflow` itself.

File: pkgs/development/interpreters/nextflow/default.nix
Comment (rollf) [inline pkgs/development/interpreters/nextflow/default.nix:32]: Okay, thanks for this @SFrijters . If I change my PR (which I'm happy to do), this will clash. Could you rather not change the nextflow derivation in your PR and simply wait till mine is merged?

Comment (SFrijters) [reply pkgs/development/interpreters/nextflow/default.nix:32]: Sure, I just put this up as a draft to get your opinion, we can proceed (or not) however you want.

This just fixes eval since openjdk doesn't cross-compile and I don't actually use this package myself (it's just a cleanup) so I would like you to check/review that it doesn't break anything for normal use when you're ready.

Comment (SFrijters) [reply pkgs/development/interpreters/nextflow/default.nix]: Thanks for the reminder, I've rebased so this PR now again only affects `nf-test` itself.

File: pkgs/by-name/nf/nf-test/package.nix
Comment (doronbehar) [inline pkgs/by-name/nf/nf-test/package.nix:27]: The committer who allowed this to enter should be punished! (@nbraud who merged https://github.com/NixOS/nixpkgs/pull/308756 , not @rollf ).

**Review by doronbehar — APPROVED** [review]

File: pkgs/by-name/nf/nf-test/package.nix
Comment (rollf) [reply pkgs/by-name/nf/nf-test/package.nix:27]: > The committer who allowed this to enter should be punished! (@nbraud who merged #308756 , not @rollf ).

That sounds a bit too rude to me.

Thanks for checking/fixing this @doronbehar @SFrijters.

**Pull request merged**
